src/ucp/api/ucp_def.h

```diff
@@ -81,6 +82,29 @@ typedef struct ucp_config                ucp_config_t;
 typedef struct ucp_ep                    *ucp_ep_h;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Callback to process incoming active message
+ *
+ * @param [in]  arg     User-defined argument.
+ * @param [in]  data    Points to the received data. 
+ * @param [in]  length  Length of data.
+ * @param [in]  flags   Ignore, these are only relevant at the UCT level
+ *
+ * @note This callback could be set and released
+ *       by @ref ucp_worker_set_am_handler function.
+ *       The minimum value of length is 8. If a user only
+ *       specifies to send 1 byte, that will be the only
+ *       valid byte in data, however length will still be set to 8.
+ *
+ * @retval UCS_OK         - The user must always specify the function to
```

Comment:
so why not make the function return void?

Reply:
Because the callback is invoked at the UCT layer, it requires that UCS_OK or UCS_INPROGRESS is returned.

Reply:
it may be handled by proxy callback in UCP

Reply:
We are switching to this approach.